Incorrect constructor result for static inner class

Consider this code,
```java
import org.checkerframework.checker.determinism.qual.*;

public class InnerClass {

    public static void f() {
        @Det C a = new C();
    }

    public static class C {
    }
}
```

This gives the following error
```
InnerClass.java:6: error: [assignment.type.incompatible] incompatible types in assignment.
        @Det C a = new C();
                   ^
  found   : @NonDet C
  required: @Det C
```

If a constructor has no `@PolyDet` arguments, its return type should be `@Det` by default but here we see this returns `@NonDet`. This only seems to occur with inner classes, with the following change,

```java
import org.checkerframework.checker.determinism.qual.*;

public class InnerClass {

    public static void f() {
        @Det InnerClass a = new InnerClass();
    }

    public static class C {
    }
}
```

I get no error.

--- a/.travis-build.sh
+++ b/.travis-build.sh
@@ -71,7 +71,7 @@ fi
 
 if [[ "${GROUP}" == "jdk.jar" || "${GROUP}" == "all" ]]; then
   ## Run the tests for the type systems that use the annotated JDK
-  ./gradlew IndexTest LockTest NullnessFbcTest OptionalTest -PuseLocalJdk --console=plain --warning-mode=all --no-daemon
+  ./gradlew DeterminismTest IndexTest LockTest NullnessFbcTest OptionalTest -PuseLocalJdk --console=plain --warning-mode=all --no-daemon
 fi
 
 if [[ "${GROUP}" == "misc" || "${GROUP}" == "all" ]]; then

--- a/checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java
+++ b/checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java
@@ -557,7 +557,8 @@ public class DeterminismAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
                     // First check receiver type. If it is unannotated or @PolyDet, don't check the
                     // rest of the parameters.
                     AnnotatedTypeMirror receiverType = executableType.getReceiverType();
-                    if (receiverType != null) {
+                    if (receiverType != null
+                            && executableType.getElement().getKind() != ElementKind.CONSTRUCTOR) {
                         unannotatedOrPolyDet =
                                 receiverType.getAnnotations().isEmpty()
                                         || receiverType.hasAnnotation(POLYDET);

--- a/checker/tests/determinism/AllTestsMultipleUnions.java
+++ b/checker/tests/determinism/AllTestsMultipleUnions.java
@@ -33,6 +33,5 @@ public class AllTestsMultipleUnions {
 
     class MyExceptionB extends Throwable implements Cloneable, MyInterface<String> {}
 
-    // :: error: (super.invocation.invalid)
     class SubMyExceptionA extends MyExceptionA {}
 }

--- a/checker/tests/determinism/AllTestsUnionCrash.java
+++ b/checker/tests/determinism/AllTestsUnionCrash.java
@@ -26,6 +26,5 @@ public class AllTestsUnionCrash {
 
     class MyExceptionB extends Throwable implements Cloneable, MyInterface<String> {}
 
-    // :: error: (super.invocation.invalid)
     class SubMyExceptionA extends MyExceptionA {}
 }

--- a/checker/tests/determinism/AllTestsUnions.java
+++ b/checker/tests/determinism/AllTestsUnions.java
@@ -43,9 +43,7 @@ public class AllTestsUnions {
 
     class MyExceptionB extends Throwable implements Cloneable, MyInterface<String> {}
 
-    // :: error: (super.invocation.invalid)
     class SubMyExceptionA extends MyExceptionA {}
 
-    // :: error: (super.invocation.invalid)
     class SubMyExceptionA2 extends MyExceptionA {}
 }

--- a/checker/tests/determinism/DetIssue114Defaults.java
+++ b/checker/tests/determinism/DetIssue114Defaults.java
@@ -1,0 +1,16 @@
+import org.checkerframework.checker.determinism.qual.*;
+
+public class DetIssue114Defaults {
+
+    public static void f() {
+        @Det DetIssue114Defaults b = new DetIssue114Defaults();
+        @Det C a = new C();
+    }
+
+    public static class C {
+        void method() {
+            @Det DetIssue114Defaults b = new DetIssue114Defaults();
+            @Det C a = new C();
+        }
+    }
+}

